src/Shared/Process/ProcessEx.cs

```diff
@@ -49,10 +49,15 @@ public ProcessEx(ITestOutputHelper output, Process proc, TimeSpan timeout)
 
             _exited = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
 
+            // We greedily create a timeout exception message even though a timeout is unlikely to happen for two reasons:
+            // 1. To make it less likely for Process getters to throw exceptions like "System.InvalidOperationException: Process has exited, ..."
+            // 2. To ensure if/when exceptions are thrown from Process getters, these exceptions can easily be observed.
+            var timeoutExMessage = $"Process proc {proc.ProcessName} {proc.StartInfo.Arguments} timed out after {DefaultProcessTimeout}.";
```

Comment:
Late questions (sorry):
1. Have you confirmed `ProcessName` is available here i.e. the process is always started before the constructor is called❔
2. Do we need a similar change in src/Tools/Microsoft.dotnet-openapi/test/ProcessEx.cs and wherever else we have similar `Process` wrappers❔ (No, I don't remember why we have more than one `ProcessEx` class.)

Reply:
1. `_process` is private and there's no mechanism in ProcessEx to start a process, so it must be started here.
2. That's probably a good idea. I didn't know there were copies of this. Is that the only other one?